homeassistant/components/hue_ble/light.py

```diff
@@ -0,0 +1,155 @@
+
+        self._api.add_callback_on_state_changed(self._state_change_callback)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from HA."""
+
+        self._api.remove_callback(self._state_change_callback)
+
+    def _update_updatable_attributes(self) -> None:
+        """Update this entities updatable attrs from the lights state."""
+        self._attr_available = self._api.available
+        self._attr_is_on = self._api.power_state
+        self._attr_brightness = self._api.brightness
+        self._attr_color_temp_kelvin = (
+            color_util.color_temperature_mired_to_kelvin(self._api.colour_temp)
+            if self._api.colour_temp is not None and self._api.colour_temp != 0
+            else -1
+        )
+        self._attr_xy_color = self._api.colour_xy
+
+    def _state_change_callback(self) -> None:
+        """Run when light informs of state update. Updates local properties."""
+        _LOGGER.debug("Received state notification from light %s", self.name)
+        self._update_updatable_attributes()
+        self.async_write_ha_state()
+
+    async def async_update(self) -> None:
+        """Fetch latest state from light and make available via properties."""
+        await self._api.poll_state(run_callbacks=True)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Set properties then turn the light on."""
+
+        _LOGGER.debug("Turning light %s on with args %s", self.name, kwargs)
+
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs[ATTR_BRIGHTNESS]
+            _LOGGER.debug("Setting brightness of %s to %s", self.name, brightness)
+            await self._api.set_brightness(brightness)
+
+        if ATTR_COLOR_TEMP_KELVIN in kwargs:
+            color_temp_kelvin = kwargs[ATTR_COLOR_TEMP_KELVIN]
+            mireds = color_util.color_temperature_kelvin_to_mired(color_temp_kelvin)
+            _LOGGER.debug("Setting color temp of %s to %s", self.name, mireds)
+            await self._api.set_colour_temp(mireds)
+
+        if ATTR_XY_COLOR in kwargs:
+            xy_color = kwargs[ATTR_XY_COLOR]
+            _LOGGER.debug("Setting XY color of %s to %s", self.name, xy_color)
+            await self._api.set_colour_xy(xy_color[0], xy_color[1])
+
+        await self._api.set_power(True)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn light off then set properties."""
+
+        _LOGGER.debug("Turning light %s off with args %s", self.name, kwargs)
+        await self._api.set_power(False)
+
+    @property
+    def color_mode(self) -> ColorMode:
+        """Color mode of the light."""
+
+        if self._api.supports_colour_xy:
+            if self._api.supports_colour_temp and self._api.colour_temp_mode:
+                return ColorMode.COLOR_TEMP
+            return ColorMode.XY
+
+        if self._api.supports_brightness:
+            return ColorMode.BRIGHTNESS
+
+        return ColorMode.ONOFF
```

Comment:
The logic is not correct, we will return `ColorMode.BRIGHTNESS` or `ColorMode.ONOFF` for a light which only supports `ColorMode.COLOR_TEMP`, which is not correct.

Please add a test for a light with support for `ColorMode.COLOR_TEMP` if there is not already such a test.